docs/hub/how-to-downstream.md

```diff
@@ -0,0 +1,51 @@
+# How to integrate downstream methods in your library
+
+In the tutorial, you learned some basic functions for integrating the Hub into your library. Functions that allow your library to download files from the Hub are referred to as *downstream* functions. This guide introduces additional downstream methods you can integrate in your library. You will learn how to:
+
+* Download a file without caching it on your disk.
+* Download a file and cache it on your disk.
+* Download all the files in a repository.
+
+## `hf_hub_url`
+
+Use `hf_hub_url` to download a specific file from a repository by providing a `filename`. This function takes the URL of the file, and downloads it. For example, if you want to download a `config.json` file from a repository:
```

Comment:
Quick explanation for context since there might be a bit of misunderstanding of this method. `hf_hub_url` only gives you a url in the Hub, it does not download the file. Usually you need the url to make a call to download the file, so that's why `cached_download` (which uses url) is usually combined with `hf_hub_url`

Maybe it could be something like this:

```suggestion
Use `hf_hub_url` to obtain the url of a specific file from a repository by providing a `filename`. This function takes the repository id the file name. For example, if you want to get the url of a `config.json` file from a repository:
```

Reply:
Thanks for clarifying!